web/src/message_view_header.ts

```diff
@@ -31,35 +53,42 @@ function get_message_view_header_context(filter) {
             zulip_icon: "all-messages",
         };
     }
-    const context = filter.add_icon_data({
-        title: filter.get_title(),
+    const title = filter.get_title();
+    assert(title !== undefined);
```

Comment:
I'm pretty sure the title can be `undefined` in any case where the narrow is too complex for it to label, like a search for `has:image has:attachment sender:me`.

Reply:
Ah okay, so `filter.add_icon_data({` should be updated then. Thanks!

Reply:
Hm, I'm not sure how to reproduce the undefined situation. When the narrow is too complex, we keep the search bar open and there's no narrow title.

Reply:
Ahh, yeah, I think the `filter.is_common_narrow` check in this code path manages that:

```
function build_message_view_header(filter) {                                                            
    // This makes sure we don't waste time appending                                                    
    // message_view_header on a template where it's never used                                          
    if (filter && !filter.is_common_narrow()) {                                                         
        search.open_search_bar_and_close_narrow_description();                                          
        search.set_search_bar_text(narrow_state.search_string());                                       
    } else {                                                                                            
        const context = get_message_view_header_context(filter);                                        
        append_and_display_title_area(context);                                                         
        search.close_search_bar_and_open_narrow_description();                                          
    }                                                                                                   
}                                                                                                       
```

Reply:
It's a bit evil how this non-local this logic interaction, but the new `assert` statement is correct from a typing perspective. 

Probably there's a good refactor here that would have `get_message_view_header_context` just return `undefined` if we're supposed to keep the input closed, and have this read as:
```
const header_context = get_message_view_header_context(filter);
if (header_context === undefined) {
    search.open_search_bar_and_close_narrow_description();
    search.set_search_bar_text(narrow_state.search_string());  
    return;
}

append_and_display_title_area(context);
search.close_search_bar_and_open_narrow_description();

```

and return `undefined` when `filter.get_title` doesn't have a title for us, so that there's just one bit of logic making the decision? I think that might have identical effect to what we have now, but without the implicit requirement that `filter.is_common_narrow` and `filter.get_title` always agree.